apps/gitness/src/pages-v2/pull-request/hooks/usePRChecksDecision.ts

```diff
@@ -0,0 +1,181 @@
+import { useEffect, useMemo, useState } from 'react'
+
+import pluralize from 'pluralize'
+
+import { TypesPullReq, TypesRepository, useChecksPullReqQuery } from '@harnessio/code-service-client'
+import { ExecutionState } from '@harnessio/views'
+
+import { determineStatusMessage, generateStatusSummary } from '../../../pages/pull-request/utils'
+
+interface Check {
+  check: {
+    status: ExecutionState
+  }
+}
+
+interface Data {
+  checks: Check[]
+}
+export function usePRChecksDecision({
+  repoMetadata,
+  pullReqMetadata
+}: {
+  repoMetadata: TypesRepository | undefined
+  pullReqMetadata: TypesPullReq | undefined
+}) {
+  const {
+    data: { body: data } = {},
+    error,
+    refetch
+  } = useChecksPullReqQuery({
+    repo_ref: `${repoMetadata?.path}/+`,
+    pullreq_number: pullReqMetadata?.number ?? 0,
+    queryParams: { debounce: 500 }
+  })
+  const [count, setCount] = useState(DEFAULT_COUNTS)
+  const [color, setColor] = useState<string>('GREEN_500')
+  const [background, setBackground] = useState<string>('GREEN_50')
+  const [message, setMessage] = useState('')
+  const [complete, setComplete] = useState(true)
+  const [summaryText, setSummaryText] = useState('')
+  const [checkInfo, setCheckInfo] = useState<{
+    title: string
+    content: string
+    color: string
+    status: string
+  }>({ title: '', content: '', color: '', status: '' })
+
+  const status = useMemo(() => {
+    let _status: ExecutionState | undefined
+    const _count = { ...DEFAULT_COUNTS }
+    const total = data?.checks?.length
+    // @ts-expect-error remove "@ts-expect-error" once CodeServiceClient Response for useChecksPullReqQuery is fixed
+    const { message: summaryMessage } = generateStatusSummary(data?.checks)
+    setSummaryText(summaryMessage)
+    // @ts-expect-error remove "@ts-expect-error" once CodeServiceClient Response for useChecksPullReqQuery is fixed
+    const checkInfoData = determineStatusMessage(data?.checks)
+    if (checkInfoData) {
+      setCheckInfo(checkInfoData)
+    }
+    if (total) {
+      for (const check of (data as Data).checks) {
+        switch (check.check.status) {
+          case ExecutionState.ERROR:
+          case ExecutionState.FAILURE:
+          case ExecutionState.RUNNING:
+          case ExecutionState.PENDING:
+          case ExecutionState.SUCCESS:
+            _count[check.check.status]++
+            setCount({ ..._count })
```

Comment:
Might be better simplifying this a bit since it currently mixes mutation of the current value with replacing.
```suggestion
            setCount({ ..._count, [check.check.status]: _count[check.check.status] + 1 })
```

Reply:
done